### PR TITLE
docs: Add Python binding build instructions

### PR DESCRIPTION
**Pull request by parker-research**
Related to discussions in https://github.com/MikePopoloski/pyslang/issues/381, https://github.com/MikePopoloski/pyslang/issues/379

### CODE REVIEW

File: docs/building.dox [hunk 1]
```diff
@@ -124,6 +124,38 @@ cmake --build build --target docs
 
 The output website is located at `build/docs/html/`
 
+@section python-binding-builds Building Python Bindings
+
+This section contains instructions for building and installing the Python bindings, which are created with [pybind11](https://github.com/pybind/pybind11).
+
+## Steps
+
+1. Clone the `pyslang` repository (https://github.com/MikePopoloski/slang)
```
Comment (MikePopoloski) [inline docs/building.dox:133]: URL points to the wrong repository.

File: docs/building.dox [hunk 2]
```diff
@@ -124,6 +124,38 @@ cmake --build build --target docs
...
+2. Clone the `slang` submodule:
+
+@code{.ansi}
+git submodule update --init --recursive
+@endcode
+
+3. Create a virtual environment, activate it, and install `pyslang` as a Python package (including building the C++ `slang` library, with bindings):
```
Comment (MikePopoloski) [inline docs/building.dox:140]: The venv isn't necessary, but I suppose it's fine to include it.

Comment (parker-research) [reply docs/building.dox:140]: Will change to indicate it's optional

**Comment by parker-research** [comment]
Implemented review suggestions!

**Pull request merged**